ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java

```diff
@@ -147,16 +151,20 @@ public void writeRecord(final ConsumerRecord<byte[], byte[]> record) {
   }
 
   void writeCommandToBackup(final ConsumerRecord<CommandId, Command> record) {
+    if (corruptionDetected) {
+      LOG.warn("Failure to write command topic data to backup. "
```

Comment:
Can we throw an exception instead of logging this message? We should probably catch the error in the CommandTopic class instead of checking for the sync error. Whoever calls the `writeCommandToBackup` will know that the issue is not recoverable, so the exception is good to have. A log message is just informative, but the old method was actually doing something with that (i.e. creating a new backup file).